src/mmap_unix.rs

```diff
@@ -74,6 +74,142 @@ impl error::Error for Error {}
 
 pub type Result<T> = result::Result<T, Error>;
 
+/// A factory struct to build `MmapRegion` objects.
+pub struct MmapRegionBuilder<B = ()> {
+    size: usize,
+    prot: i32,
+    flags: i32,
+    file_offset: Option<FileOffset>,
+    raw_ptr: Option<*mut u8>,
+    hugetlbfs: Option<bool>,
+    bitmap: B,
+}
+
+impl<B: Bitmap> MmapRegionBuilder<B> {
+    /// Create a new `MmapRegionBuilder`.
```

Comment:
Should we add a comment mentioning that the `bitmap` parameter needs to be `()` if the user doesn't want the dirty page tracking?

Reply:
Added a comment about that.